build_tools/docker/cmake_swiftshader/Dockerfile

```diff
@@ -0,0 +1,59 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# An image for building IREE using CMake and testing IREE with SwiftShader
+# Vulkan implementation.
+
+# Build using:
+# docker build --tag gcr.io/iree-oss/cmake-swiftshader \
+#              build_tools/docker/cmake_swiftshader/
+
+# Run interactively using the following, where IREE_WORKDIR is the path to your
+# local dev environment:
+# docker run -it --rm --entrypoint bash \
+#   --volume "${IREE_WORKDIR?}:/usr/src/iree/" \
+#   --gpus all \
+#   gcr.io/iree-oss/cmake-swiftshader
+
+# Set up the image and working directory by inheriting the base
+# CMake configuration.
+FROM gcr.io/iree-oss/cmake-vulkan
+
+ARG SWIFTSHADER_COMMIT=master
+
+# zlib is needed for compiling SwiftShader.
+RUN apt-get update && apt-get install zlib1g-dev
+
+RUN git clone https://github.com/google/swiftshader \
```

Comment:
Should we use a specific commit here? Maybe the version used in the IREE submodule? Or maybe use https://sourcegraph.com/github.com/google/iree/-/blob/build_tools/third_party/swiftshader/build_vk_swiftshader.sh to use our submodule.

Reply:
The commit tracked by IREE is quite old; I suspect it's causing test failures so I fetched this commit (originally planned in the next patch) to test it out. I was chatting with @ScottTodd and he suggested that we might want to remove SwiftShader as a submodule dependency given it's really an Vulkan ICD for testing purpose. IREE itself does not depend on it for building or whatever. Developers needing it should be able to build by themselves. (We can still keep the build script for convenience I guess.)

Reply:
SwiftShader does not have releases or versions at the moment. So we are basically random selecting one SHA anyway.. I've logged the SHA used in `/swiftshader/git-commit` and we can override this `ARG` if necessary.

Reply:
That all makes sense to me. I think we should still specify the SHA to avoid it updating mysteriously when updating the container for another reason and Docker decides it needs to rebuild that layer. Can we make it an ARG?

Reply:
Yeah make sense. Done.